symforce/codegen/codegen_config.py

```diff
@@ -29,58 +33,37 @@ class CodegenConfig:
     cse_optimizations: T.Optional[
         T.Union[T.Literal["basic"], T.Sequence[T.Tuple[T.Callable, T.Callable]]]
     ] = None
+    # TODO(hayk): Remove this parameter (by making everything 2D?)
+    matrix_is_1d: bool = False
 
+    @classmethod
+    def backend_name(cls) -> str:
+        """
+        String name for the backend. This should match the directory name in codegen/backends
+        and will be used to namespace by backend in generated code.
+        """
+        raise NotImplementedError()
 
-@dataclass
-class CppConfig(CodegenConfig):
-    """
-    C++ Codegen configuration
-
-    Args:
-        doc_comment_line_prefix: Prefix applied to each line in a docstring
-        line_length: Maximum allowed line length in docstrings; used for formatting docstrings.
-        use_eigen_types: Use eigen_lcm types for vectors instead of lists
-        autoformat: Run a code formatter on the generated code
-        cse_optimizations: Optimizations argument to pass to sm.cse
-        support_complex: Generate code that can work with std::complex or with regular float types
-        force_no_inline: Mark generated functions as `__attribute__((noinline))`
-        zero_initialization_sparsity_threshold: Threshold between 0 and 1 for the sparsity below
-                                                which we'll initialize an output matrix to 0, so we
-                                                don't have to generate a line to set each zero
-                                                element to 0 individually
-        explicit_template_instantiation_types: Explicity instantiates templated functions in a `.cc`
-            file for each given type. This allows the generated function to be compiled in its own
-            translation unit. Useful for large functions which take a long time to compile.
-    """
+    @classmethod
+    def template_dir(cls) -> Path:
+        """
+        Directory for jinja templates.
+        """
+        raise NotImplementedError()
 
-    doc_comment_line_prefix: str = " * "
-    line_length: int = 100
-    use_eigen_types: bool = True
-    support_complex: bool = False
-    force_no_inline: bool = False
-    zero_initialization_sparsity_threshold: float = 0.5
-    explicit_template_instantiation_types: T.Optional[T.Sequence[str]] = None
+    def templates_to_render(self, generated_file_name: str) -> T.List[T.Tuple[str, str]]:
+        raise NotImplementedError()
 
+    def printer(self) -> "sm.CodePrinter":
```

Comment:
Made it sympy. I think part of the reason this was in quotes was to avoid importing SymPy into the config files. We could import sympy in an `if T.TYPE_CHECKING` block, but that's a bit ugly. Do you think we should do that?

Reply:
It's probably fair to assume if you're importing the codegen config you're going to generate code, so you'll need to import sympy anyway - I'd lean towards just importing sympy here, and if there's a use case where that's annoyingly slow we can move it to a `T.TYPE_CHECKING` block

Reply:
Okay I'm on board. It looks like it should be `sympy.printing.codeprinter.CodePrinter` though?